editor.planx.uk/src/ui/shared/CharacterCounter.tsx

```diff
@@ -0,0 +1,92 @@
+import Typography from "@mui/material/Typography";
+import { visuallyHidden } from "@mui/utils";
+import { TEXT_LIMITS, TextInputType } from "@planx/components/TextInput/model";
+import { debounce } from "lodash";
+import React, { useCallback, useEffect, useState } from "react";
+import { FONT_WEIGHT_SEMI_BOLD } from "theme";
+
+export type Props = {
+  textInputType: TextInputType;
+  count: number;
+  error: boolean;
+};
+
+export function isLongTextType(
+  type: TextInputType,
+): type is TextInputType.Long | TextInputType.ExtraLong {
+  return type === TextInputType.Long || type === TextInputType.ExtraLong;
+}
+
+export const CharacterCounter: React.FC<Props> = ({
+  textInputType,
+  count,
+  error,
+}) => {
+  const [screenReaderCount, setScreenReaderCount] = useState<number>(0);
+  const [showReaderCount, setShowReaderCount] = useState<boolean>(false);
+
+  const updateCharacterCount = useCallback(
+    debounce((count: number) => {
+      setScreenReaderCount(count);
+      setShowReaderCount(true);
+    }, 500),
+    [],
+  );
+
+  function getLongTextLimit(type: TextInputType): number {
+    if (isLongTextType(type)) {
+      return TEXT_LIMITS[type].limit;
+    } else {
+      return 0;
```

Comment:
return a 0 for the situation where nothing happens, but assuming there's so much type guarding and logic before this, it feels like an unlikely situation